web/src/main/java/org/fao/geonet/monitor/link/LinkCheckerWps.java

```diff
@@ -0,0 +1,170 @@
+package org.fao.geonet.monitor.link;
+
+import au.com.bytecode.opencsv.CSVReader;
+import org.apache.log4j.Logger;
+import org.jdom.Element;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class LinkCheckerWps extends LinkCheckerDefault {
+
+    private static Logger logger = Logger.getLogger(LinkCheckerWps.class);
+    private static String WFS_GET_FEATURES = "?service=WFS&version=1.0.0&request=GetFeature&maxFeatures=1&outputFormat=csv&typeName=";
+    private String wfsUrl;
+    private String layer;
+
+    @Override
+    public void setOnlineResource(String uuid, final Element onlineResource) {
+        super.setOnlineResource(uuid, onlineResource);
+        layer = LinkCheckerUtils.parseOnlineResource(onlineResource, LinkCheckerUtils.NAME_XPATH);
+        wfsUrl = url + WFS_GET_FEATURES + layer;
+        url += "?service=WPS";
+    }
+
+    @Override
+    public boolean check() {
+
+        try {
+            String time = getTime();
+            if (time == null)
+                return false;
+
+            String requestXml = getRequestXml(time);
+
+            HttpURLConnection connection;
+            connection = (HttpURLConnection) (new URL(url)).openConnection();
+            connection.setConnectTimeout(LinkMonitorService.timeout * 1000);
+            connection.setReadTimeout(LinkMonitorService.timeout * 1000);
+            connection.setRequestMethod("POST");
+            connection.setRequestProperty("Content-Type", "text/xml");
+            connection.setDoOutput(true);
+
+            OutputStreamWriter writer = new OutputStreamWriter(connection.getOutputStream());
+            writer.write(requestXml);
+            writer.flush();
+            writer.close();
```

Comment:
Does this need to be wrapped, to guarantee the writer is closed if anything throws an exception?